### DISCUSSION TITLE
Configuration

### DISCUSSION THREAD

**Discussion by bcholmes** [LVerhulst4321/PlanZ#53] [Ideas]
There are 4 ways of configuring the app:

1. db_name.php -- this mechanism blends a bunch of different types of "switches": con name and dates, turning behaviour on and off, resource configuration, etc.
2. Phases -- some cons just don't use some phases, and some phases are only available for certain parts of the con. There's often a relationship between Phases and permissions
3. Config Tables -- certain parts of the app are configured using the config tables: rooms, divisions, etc.
4. Code -- in the past, cons made a lot of custom configurations. It'd be nice if we could understand the needs of various cons better so that a con didn't feel the need to implement a particular requirement; that it was available and just needed to be turned on and configured.

I feel like that's a lot of complexity, and I'd love to see some streamlining there.

**Comment by lostcarpark** [comment]
I would add a fifth - permissions. Access to particular functions for particular groups can be configured by changing the permission tables. As far as I know, this can only be done through SQL.

There's also the "Edit Custom Text", which I think belongs in the same category as Config Tables, but is implemented as a separate function.

Not sure if it's been tidied up for PlanZ, but in Zambia I don't think the survey is integrated into the menu by default, so it's up to cons to edit the header to add a menu entry for it. I think it would make sense to add Survey to Phases, so it could be turned on when needed, and turned when no more survey responses are required.

**Comment by LVerhulst4321** [comment]
Yeah, there's a lot of different configuration things. Streamlining that would be a pretty big project.

Regarding Survey in the menu - yes, it is now integrated by default. If USING_SURVEYS is true in the db_name.php, then the menu option appears.

**Comment by lostcarpark** [comment]
Oh yes, I did notice that setting!

It does strike me as something that would fit well with phases. I'm a fan of giving the programme team the ability to control as much as possible without bothering the IT team.

On the other had, if we were going to drop one configuration method, phases would probably be the most obvious choice.

**Comment by LVerhulst4321** [comment]
Drop Phases? And replace it how? I mean, it's technically a configuration table with a special screen, so I doubt that you want to just make it a normal config table. So, how would the staff/admin turn on/off certain functionality?

I am more in favor of moving as many items out of the db_name.php file as possible and into a config table to allow better control by the PlanZ staff/admin and not having to rely on an IT person. Survey is one of those and making it a Phase item sounds good to me.

**Reply by bcholmes** [reply]
Yeah, I'm with you on that. I'm not quite understanding James' suggestion.

**Reply by lostcarpark** [reply]
Sorry, it wasn't a serious suggestion, just if there was a config mechanism to get rid of it would seem the easiest choice.

I don't think we could get rid of db_name, but I do think that it should be reserved for "technical" stuff like database and email server configuration. Being able to give programme team control of anything that doesn't have a technical reason for being under the control of IT staff would make sense.

One things that springs to mind is the timeslots. Control of timeslots for scheduling is in db_name, while control of the timeslots for availability is in a config table. I think it would make sense to have one set of timeslots for both, and it ought to be under the control of the programme team.
